### PR TITLE
Save and update query params based on editor state

### LINKED ISSUE #75: Read content & preferences from URL query string
This would make sharing links to the page much easier.

For example, `https://carbon.now.sh/?code=Hello%20world!&theme=blackboard` would pre-fill the editor with "Hello world!" and apply the *Blackboard* theme. Everything else would fall back to the default settings, of course.

If this would be a welcome addition, I'll open a PR for it.

### PR DESCRIPTION
**Pull request by mfix22**
Update editor to save and load editor state from query parameters (e.g. `?code=Hello&l=javascript`)

Please help review and test 😄 

- Closes #75

### CODE REVIEW

File: pages/editor.js [hunk 1]
```diff
@@ -23,51 +24,64 @@ import {
...
+        dropShadow: true,
+        windowControls: true,
+        widthAdjustment: true,
+        paddingVertical: '48px',
+        paddingHorizontal: '32px',
+        uploading: false,
+        code: props.content,
+        _initialState: this.props.intialState
+      },
+      this.props.intialState
```
Comment (yeskunall) [inline pages/editor.js]: Could just use the spread syntax here.

File: pages/editor.js [hunk 2]
```diff
@@ -23,51 +24,64 @@ import {
...
+        theme: THEMES.seti.id,
+        language: DEFAULT_LANGUAGE,
+        dropShadow: true,
+        windowControls: true,
+        widthAdjustment: true,
+        paddingVertical: '48px',
+        paddingHorizontal: '32px',
+        uploading: false,
+        code: props.content,
+        _initialState: this.props.intialState
```
Comment (yeskunall) [inline pages/editor.js]: Typo: should probably be `initialState`. L62 and L64.

Comment (mfix22) [reply pages/editor.js]: Great catch! I might have never seen that!

File: pages/editor.js [hunk 3]
```diff
@@ -23,51 +24,70 @@ import {
...
+    if (!this.state._initialState) {
+      const state = getState(localStorage)
+      if (state) {
+        this.setState(state)
+      }
     }
   }
 
   componentDidUpdate() {
+    const mappedState = mapper.map(reverseMappings, this.state)
```
Comment (briandennis) [inline pages/editor.js]: This should probably be done inside of `replaceQuery` so all that the editor component needs to do is pass a state object to a function. Also maybe rename to `updateQueryString` or something to make its use clear?

Comment (briandennis) [reply pages/editor.js]: Or another way, editor probably shouldn't have to know about `mapper` or `reverseMappings`

**Pull request merged**